agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/sql/PreparedStatementClassDataProvider.java

```diff
@@ -230,8 +231,23 @@ public ClassVisitor create(ClassInstrumentationData classInstrumentationData, Cl
                 return new PreparedStatementClassVisitor(classInstrumentationData, classWriter, metaData1);
             }
         };
+    }
 
-        return classVisitorFactory;
+    private ClassVisitorFactory classFactoryForNewPostgreSql() {
+        return new ClassVisitorFactory() {
+            @Override
+            public ClassVisitor create(ClassInstrumentationData classInstrumentationData, ClassWriter classWriter) {
+                HashSet<String> ctorSignatures = new HashSet<String>();
```

Comment:
In general using interface is preferred.  I would like to see the change:

`Set<String> ctorSignatures = new HashSet<>();`